src/cbdata.cc

```diff
@@ -89,6 +90,14 @@ class cbdata
     {}
     ~cbdata();
 
+    static cbdata *FromUserData(const void *p) {
+#if WITH_VALGRIND
+        return cbdata_htable.at(p);
+#else
+        return (cbdata *) (((char *) p) - offsetof(cbdata, data));
```

Comment:
If you get a chance, please use appropriate C++ casts here.

Reply:
Done in 207b801